compilation failure on Windows due to smimesign dependency

### What's wrong?

I wanted to try and debug some test failures on Windows so I `git clone`d the repo and tried to run `go test` but it failed with errors around our `github.com/grafana/smimesign` dependency.

I see in our smimesign fork that the functions mentioned [do exist](https://github.com/grafana/smimesign/blob/main/certstore/certstore_windows.go) on certstore_windows.go, but I don't see any build tags other than the weird `#cgo windows LDFLAGS: -lcrypt32 -lncrypt` line.

Is there some other dependency I'm supposed to be installing that we should document?

Also, here's my `go env` in case it matters

<details>

```
$ go env
set GO111MODULE=
set GOARCH=amd64
set GOBIN=
set GOCACHE=C:\Users\ookam\AppData\Local\go-build
set GOENV=C:\Users\ookam\AppData\Roaming\go\env
set GOEXE=.exe
set GOEXPERIMENT=
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOINSECURE=
set GOMODCACHE=C:\Users\ookam\go\pkg\mod
set GONOPROXY=
set GONOSUMDB=
set GOOS=windows
set GOPATH=C:\Users\ookam\go
set GOPRIVATE=
set GOPROXY=https://proxy.golang.org,direct
set GOROOT=C:\Program Files\Go
set GOSUMDB=sum.golang.org
set GOTMPDIR=
set GOTOOLCHAIN=auto
set GOTOOLDIR=C:\Program Files\Go\pkg\tool\windows_amd64
set GOVCS=
set GOVERSION=go1.22.0
set GCCGO=gccgo
set GOAMD64=v1
set AR=ar
set CC=gcc
set CXX=g++
set CGO_ENABLED=0
set GOMOD=C:\Users\ookam\GitRepos\agent\go.mod
set GOWORK=
set CGO_CFLAGS=-O2 -g
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-O2 -g
set CGO_FFLAGS=-O2 -g
set CGO_LDFLAGS=-O2 -g
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -fno-caret-diagnostics -Qunused-arguments -Wl,--no-gc-sections -fmessage-length=0 -ffile-prefix-map=C:\Users\ookam\AppData\Local\Temp\go-build2325116132=/tmp/go-build -gno-record-gcc-switches
```

</details>

### Steps to reproduce

Clone the repo, try to build/test the agent after just installing Git and Go.

### System information

Windows 11N

### Software version

main (91ea77d3311a6f64c8f690fd89916ad4bb783826)

### Configuration

_No response_

### Logs

```text
$ git clone https://github.com/grafana/agent
$ cd agent/pkg/flow
$ go test ./...
# github.com/github/smimesign/certstore
..\..\..\..\go\pkg\mod\github.com\grafana\smimesign@v0.2.1-0.20220408144937-2a5adf3481d3\certstore\certstore.go:17:9: undefined: openStore
..\..\..\..\go\pkg\mod\github.com\grafana\smimesign@v0.2.1-0.20220408144937-2a5adf3481d3\certstore\certstore.go:20:34: undefined: StoreType
..\..\..\..\go\pkg\mod\github.com\grafana\smimesign@v0.2.1-0.20220408144937-2a5adf3481d3\certstore\certstore.go:21:9: undefined: openSpecificStore
FAIL    github.com/grafana/agent/pkg/flow [build failed]
```

--- a/docs/developer/contributing.md
+++ b/docs/developer/contributing.md
@@ -88,6 +88,11 @@ Compiling Grafana Agent on Linux requires extra dependencies:
 * [systemd headers](https://packages.debian.org/sid/libsystemd-dev) for Promtail
    * Can be installed on Debian-based distributions with: ```sudo apt-get install libsystemd-dev```
 
+### Compile on Windows
+Compiling Grafana Agent on Windows requires extra dependencies:
+
+* [tdm-gcc](https://jmeubank.github.io/tdm-gcc/download/) full 64 bit install for cgo compiliation.
+
 ## Pull Request Checklist
 
 Changes should be branched off of the `main` branch. It's recommended to rebase

